### ISSUE TITLE
Replace block.difficulty with block.prevrandao

### ISSUE THREAD

**Issue by sdn222** [erc721r/ERC721R#11]
Compiler generates the following warning:

![image](https://github.com/erc721r/ERC721R/assets/114336669/c881d1f9-d572-4447-981c-f5f10bc5121a)

**Comment by RogerPodacter** [comment]
Yes this is a good point, it shouldn't generate a warning. The nice thing about the current situation though is that it will work on 0.8.17 and below where `prevrandao` doesn't exist.

Given so many things use 0.8.17, I think we should support it, so what about changing this to `blockhash(block.number - 1)`? I think it's basically the same as `prevrandao` for this situation. Thoughts?

**Comment by sdn222** [comment]
The use of `blockhash` may make the outcome more predictable. So I would recommend either utilize `block.prevrandao` or stick with `block.difficulty` for now.

**Comment by RogerPodacter** [comment]
Yes, good call! Let's stick with block.difficulty for now. Hopefully people
start upgrading from 0.8.17! Thanks for raising this!
